### PR TITLE
CSV download feature

### PR DESCRIPTION
**Pull request by thomasjthomasj**
- [x] Is this Pull Request ready for review?
- [x] Do the tests pass?
- [x] Have the docs been updated?
- [ ] Have any new settings been added?  Have they got comments?  Have they been added to the settings documentation page?

**What does this Pull Request do?**
Adds a `csv_download_response()` function to the `utils` module and also dockerises the project

**Are there new tests?**
Yes, I added a basic test for the CSV response. I also had to tweak the settings file as I split it into base/local and the change of location for the settings file caused the templating tests to fail. There's a couple other changes to the settings file that fix some errors I was getting when trying to run the test application.

**Any background context you want to provide?**
This is a feature that we've used a couple times in Mcclane and it seems pointless to keep rewriting the same code.

**Where should the reviewer start?**
Check the code and that you agree with the design of it, and then try getting the app up and running following the instructions in the readme. To test the feature itself, go to http://127.0.0.1:11091/csv-link

**Additional questions or comments**
Merry Christmas

### CODE REVIEW

**Comment by johnpooch** [comment]
Looks kool to me

File: .docker/deploy/start.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+echo Installing any changed requirements...
+.docker/deploy/install_requirements.sh $ENV
+
+echo Waiting for DBs...
+.docker/deploy/wait-for-it.sh gravity.mysql:3306 --timeout=30 -- echo "Gravity DB is up."
```
Comment (currycoder) [inline .docker/deploy/start.sh]: uwotm8?

File: .docker/deploy/start.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+echo Running gulp...
+node_modules/gulp-cli/bin/gulp.js watch &
```
Comment (currycoder) [inline .docker/deploy/start.sh]: is this necessary?

File: docker-compose.override.yml.example [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  gndjango.web:
+    volumes:
+      - .:/code
+    user: "${DOCKER_UID}:${DOCKER_UID}"
+    build:
+      args:
+        - ENV=dev
+        - DOCKER_UID=${DOCKER_UID}
+    ports:
+      - "127.0.0.1:11091:8000"
```
Comment (currycoder) [inline docker-compose.override.yml.example]: GG - just to check the 110- prefix isn't used by another project?

File: gn_django/utils.py [hunk 4]
```diff
@@ -50,3 +52,34 @@ def htmlify_content(content):
...
     )
     return '<br />'.join(content.split('\n'))
+
+def csv_download_response(columns, data, filename, include_date=True):
+    """
+    Put data into a CSV download response
+
+    Args:
+        - ``columns`` - A tuple of column headings
+        - ``data`` - An iterable of data. Each value should be tuple of values aligning with
```
Comment (currycoder) [inline gn_django/utils.py]: Missing filename and include_date bullets here

File: tests/gn_django_tests/test_csv.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        data = (
+            ('Byte', '& Barq'),
+            ('Dr', 'Coyle'),
+            ('Hedlok', ''),
+            ('Helix', ''),
+        )
+        response, writer = csv_download_response(('First Name', 'Second Name'), data, 'arms-roster')
+        expected_content = b"First Name,Second Name\r\nByte,& Barq\r\nDr,Coyle\r\nHedlok,\r\nHelix,\r\n"
+        self.assertEqual(response['Content-Type'], 'text/csv')
+        self.assertEqual(response.content, expected_content)
```
Comment (currycoder) [inline tests/gn_django_tests/test_csv.py]: Would be good to also append to the writer here and make sure that it shows up in the response, right?

File: tests/gn_django_tests/test_csv.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+        self.assertEqual(writer.__class__.__module__, '_csv')
+        self.assertEqual(writer.__class__.__name__, 'writer')
```
Comment (johnpooch) [inline tests/gn_django_tests/test_csv.py]: Just a suggestion: should you test that the function behaves as expected when the data is formatted incorrectly? e.g: 

```
        data = (
            ('Byte'),  # <-----
            ('Dr', 'Coyle'),
            ('Hedlok', ''),
            ('Helix', ''),
        )
```

**Pull request merged**